scripts/wallet.js

```diff
@@ -215,25 +222,30 @@ export class Wallet {
         if (this.#ownAddresses.has(address)) {
             return this.#ownAddresses.get(address);
         }
-        const last = getNetwork().lastWallet;
-        this.#addressIndex =
-            this.#addressIndex > last ? this.#addressIndex : last;
-        if (this.isHD()) {
-            for (let i = 0; i <= this.#addressIndex + MAX_ACCOUNT_GAP; i++) {
-                const path = this.getDerivationPath(0, i);
-                const testAddress = await this.#masterKey.getAddress(path);
-                if (address === testAddress) {
-                    this.#ownAddresses.set(address, path);
-                    return path;
+        // We only check 0, 1 because that's what blockbook does too by default
+        // https://github.com/trezor/blockbook/blob/master/docs/api.md#get-xpub
+        for (const nReceiving of [0, 1]) {
+            const maxIndex = Math.max(
```

Comment:
update `addressIndices ` with the value given by the network (if `addressIndices.at(nReceiving)` is smaller)

Reply:
I feel like this shouldn't be in `isOwnAddress`, besides `getNewAddress` does this too

Reply:
ok but then `addressIndices` will not be updated until you generate a new address